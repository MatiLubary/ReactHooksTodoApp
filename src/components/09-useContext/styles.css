* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

.navbar-dark .navbar-nav .nav-link:focus, .navbar-dark .navbar-nav .nav-link:hover {
    color: red;
}
.navbar-dark .navbar-nav .activeLink {
    color: red;
}