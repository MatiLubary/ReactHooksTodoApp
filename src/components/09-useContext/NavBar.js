import React from 'react'
import {  NavLink } from "react-router-dom"
import "./styles.css"


export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <div className="container-fluid">
          <NavLink exact to="/" className="navbar-brand" >useContext</NavLink>
          <div className="collapse navbar-collapse" id="navbarNav">
            <ul className="navbar-nav">
              <li className="nav-item">
                <NavLink activeClassName="activeLink" className="nav-link" aria-current="page" exact to="/">Home</NavLink>
              </li>
              <li className="nav-item">
                <NavLink activeClassName="activeLink" className="nav-link" exact to="./about">About</NavLink>
              </li>
              <li className="nav-item">
                <NavLink activeClassName="activeLink" className="nav-link" exact to="./login">Login</NavLink>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    )
}
