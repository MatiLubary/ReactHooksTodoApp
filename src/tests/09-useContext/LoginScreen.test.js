import React from 'react';
import {mount} from 'enzyme'
import { LoginScreen } from '../../components/09-useContext/LoginScreen';
import { UserContext } from '../../components/09-useContext/UserContext';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });


describe('test with <LoginScreen />', () => {

    const setUser = jest.fn()

    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    )
    test('should display properly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should execute setUser with the expected argument', () => {
        
        wrapper.find("button").prop("onClick")();

        expect(setUser).toHaveBeenCalledWith({
            id: 123,
            name: "mati"
        })
    })
    
    
    
})
