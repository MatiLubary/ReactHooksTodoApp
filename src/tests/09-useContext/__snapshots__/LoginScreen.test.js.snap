// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with <LoginScreen /> should display properly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
